pandas/tests/indexes/datetimes/test_misc.py

```diff
@@ -172,8 +175,9 @@ def test_normalize(self):
 
 class TestDatetime64(object):
 
-    def test_datetimeindex_accessors(self):
-
+    # GH 12806
+    @pytest.mark.parametrize('time_locale', tm.get_locales())
+    def test_datetimeindex_accessors(self, time_locale):
```

Comment:
can u split out the month_name and weekday_name to a separate test
skip if number of locales is 0

can then check that we don’t skip on build where we should have a locales defined